packagehandlers/packagehandlers_test.go

```diff
@@ -34,123 +31,191 @@ type pipPackageRegexTest struct {
 	expectedRequirement string
 }
 
-// Go
-func TestGoPackageHandler_UpdateDependency(t *testing.T) {
-	goPackageHandler := GoPackageHandler{}
-	testcases := []dependencyFixTest{
+func TestUpdateDependency(t *testing.T) {
+	testCases := [][]dependencyFixTest{
+		// Go test cases
 		{
-			vulnDetails: &utils.VulnerabilityDetails{
-				SuggestedFixedVersion:       "0.0.0-20201216223049-8b5274cf687f",
-				IsDirectDependency:          false,
-				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "golang.org/x/crypto"},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "0.0.0-20201216223049-8b5274cf687f",
+					IsDirectDependency:          false,
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "golang.org/x/crypto"},
+				},
+				fixSupported:          true,
+				uniqueChecksExtraArgs: []string{"go.mod"},
+			},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "1.7.7",
+					IsDirectDependency:          true,
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "github.com/gin-gonic/gin"},
+				},
+				fixSupported:          true,
+				uniqueChecksExtraArgs: []string{"go.mod"},
```

Comment:
uniqueChecksExtraArgs : Why is this needed if we only use in go to set the go.mod in golang?

Reply:
because the test function is unifying all package managers I wanted to make it general as much as possible so I dont want to add in the general function spacial case for Go.
also when I implement it like this it makes it more sense and orginized for future addition for other package managers without overloading the main function which is long anyway